### DISCUSSION TITLE
Default Header/Column name when csv file has no header

### DISCUSSION THREAD

**Discussion by davidlukas** [JoshClose/CsvHelper#1625] [Q&A]
Hi,

I'm reading .csv files without headers. The csvhelper configuration for header:
HasHeaderRecord = false,

The files are read as dynamic records, since the amount of columns/types vary.
var data = csvReader.GetRecords<dynamic>().ToList();

CsvHelper creates these dynamic records, setting the field/column name as Field1, Field2 and so forth. Is there a way to give the fieldname/column a different default name like sfx1, sfx2 ?

**Comment by JoshClose** [comment]
```cs
csv.Configuration.GetDynamicPropertyName = (context, fieldIndex) => $"sfx{fieldIndex + 1}";
```
